int8 direction_left
int8 direction_right